src/main/java/jude/Parser.java

```diff
@@ -0,0 +1,281 @@
+            boolean valid = false;
+            String[] dateTimeFormats = {
+                "yyyy-MM-dd",
+                "yyyy-MM-dd H:mm",
+                "yyyy-MM-dd h:mm a",
+                "d MMM yyyy",
+                "d MMM yyyy H:mm",
+                "d MMM yyyy h:mm a",
+                "MMM d yyyy",
+                "MMM d yyyy H:mm",
+                "MMM d yyyy h:mm a",
+            };
+            boolean[] isDateFormats = { true, false, false, true, false, false, true, false, false };
+            for (int i = 0; i < dateTimeFormats.length; i++) {
+                // Solution below adapted from
+                // https://stackoverflow.com/questions/44925840/
+                // java-time-format-datetimeparseexception-text-could-not-be-parsed-at-index-3
+                DateTimeFormatter dateFormat = new DateTimeFormatterBuilder()
+                        .parseCaseInsensitive()
+                        .appendPattern(dateTimeFormats[i])
+                        .toFormatter(Locale.ENGLISH);
+                try {
+                    if (isDateFormats[i]) {
+                        // Solution below adapted from
+                        // https://stackoverflow.com/questions/27454025/
+                        // unable-to-obtain-localdatetime-from-temporalaccessor-when-parsing-localdatetime
+                        dateObject = LocalDate.parse(date, dateFormat).atStartOfDay();
+                    } else {
+                        dateObject = LocalDateTime.parse(date, dateFormat);
+                    }
+                    valid = true;
+                    break;
+                } catch (DateTimeParseException ignored) {
+                    // invalid date, try another format
+                }
+            }
+            if (!valid) {
+                throw new IllegalCommandException("Please input a valid date, e.g. 21 Aug 2022, "
+                        + "Aug 21 2022 or 2022-08-21.");
+            }
+        }
+        return dateObject.format(DateTimeFormatter.ofPattern("dd MMM yyyy HH:mm"));
+    }
+
+    /**
+     * Returns whether the command terminates the program.
+     *
+     * @param command The user input command.
+     * @return true if the command terminates the program, i.e. bye command, false otherwise.
+     */
+    public boolean isTerminationCommand(String command) {
+        return command.equals("bye");
+    }
+
+    /**
+     * Parses the command provided and executes it. The return value will be the response by the
+     * chatbot.
+     *
+     * @param command The command which will be provided to the parser.
+     * @return true if the command terminates the program, i.e. bye command, false otherwise.
+     * @throws IOException When system I/O fails.
+     */
+    public String parse(String command) throws IOException {
+        command = command.trim(); // to remove unintended trailing spaces
+        String[] tokens = command.split(" ", 2);
+        String response = "";
+        try {
```

Comment:
You wrote this function to parse and execute the command provided. However, as a result, it is really long and difficult to read the different cases. Perhaps you could consider refactoring out the different execution codes to separate functions so that it is much cleaner and readable.

Reply:
I agree with this. In addition, line 117 contains a redundant comment which I should remove.